pkg/resourcemonitor/resourcemonitor.go

```diff
@@ -92,30 +92,34 @@ type resourceMonitor struct {
 	nodeName          string
 	args              Args
 	podResCli         podresourcesapi.PodResourcesListerClient
+	k8sCli            kubernetes.Interface
 	topo              *ghw.TopologyInfo
 	coreIDToNodeIDMap map[int]int
 	nodeCapacity      perNUMAResourceCounter
 	nodeAllocatable   perNUMAResourceCounter
 }
 
-func NewResourceMonitor(podResCli podresourcesapi.PodResourcesListerClient, args Args) (*resourceMonitor, error) {
-	topo, err := ghw.Topology(ghw.WithPathOverrides(ghw.PathOverrides{
-		"/sys": args.SysfsRoot,
-	}))
-	if err != nil {
-		return nil, err
+func NewResourceMonitor(podResCli podresourcesapi.PodResourcesListerClient, args Args, options ...func(*resourceMonitor)) (*resourceMonitor, error) {
+	rm := &resourceMonitor{
+		podResCli: podResCli,
+		args:      args,
+	}
+	for _, o := range options {
```

Comment:
I for one actually like the single-letter identifiers, part of the UNIX heritage and actually good enough in very limited scope like this. But they also turned hard to grep, so let's try something a bit longer:
```
for _, opt := range options {
   ...
```